File: gdb/domain/current_account.py

from gdb.domain.abstract_account import AbstractAccount
from gdb.exceptions import InsufficientBalanceException

class CurrentAccount(AbstractAccount):
    def __init__(self, account_number, name, age, balance, status, pin, overdraft_limit):
        super().__init__(account_number, name, age, balance, "CURRENT", status, pin)
        
    def process_debit(self, amount: float) -> None:
        self.balance -= amount

File: src/com/gdb/domain/CurrentAccount.java
package com.gdb.domain;
import com.gdb.exceptions.*;

public class CurrentAccount extends AbstractAccount {
    private double overdraftLimit;

    public CurrentAccount(String accountNumber, String name, int age, double balance, String status, String pin, double overdraftLimit) {
        super(accountNumber, name, age, balance, "CURRENT", status, pin);
        this.overdraftLimit = overdraftLimit;
    }

    public double getOverdraftLimit() { return overdraftLimit; }
    public void setOverdraftLimit(double overdraftLimit) { this.overdraftLimit = overdraftLimit; }

    @Override
    protected void processDebit(double amount) throws AccountException {
        if (amount > (this.balance + overdraftLimit)) {
            throw new InsufficientBalanceException("Exceeds overdraft limit");
        }
        this.balance -= amount;
    }
}
